03_Modules/Certificates/src/module/module.ts

```diff
@@ -312,27 +323,131 @@ export class CertificatesModule extends AbstractModule {
   }
 
   @AsHandler(CallAction.DeleteCertificate)
-  protected _handleDeleteCertificate(
+  protected async _handleDeleteCertificate(
     message: IMessage<DeleteCertificateResponse>,
     props?: HandlerProperties,
-  ): void {
+  ): Promise<void> {
     this._logger.debug('DeleteCertificate received:', message, props);
+    // if response is successful, then trigger the get installed certificates ids
+    if (message.payload.status === DeleteCertificateStatusEnumType.Accepted) {
+      try {
+        await this.sendCall(
+          message.context.stationId,
+          message.context.tenantId,
+          CallAction.GetInstalledCertificateIds,
+          {} as GetInstalledCertificateIdsRequest,
+        );
+      } catch (e: any) {
+        console.error(
+          `There was an error sending GetInstalledCertificateIds message to the charger after installing certificate ${e.message}`,
+          e,
+        );
+      }
+    }
   }
 
   @AsHandler(CallAction.GetInstalledCertificateIds)
-  protected _handleGetInstalledCertificateIds(
+  protected async _handleGetInstalledCertificateIds(
     message: IMessage<GetInstalledCertificateIdsResponse>,
     props?: HandlerProperties,
-  ): void {
+  ): Promise<void> {
     this._logger.debug('GetInstalledCertificateIds received:', message, props);
+    const certificateHashDataList: CertificateHashDataChainType[] =
+      message.payload.certificateHashDataChain!;
+    // persist installed certificate information
+    if (certificateHashDataList && certificateHashDataList.length > 0) {
+      // delete previous hashes for station
+      try {
+        await this._installedCertificateRepository.deleteAllByQuery({
```

Comment:
We cannot always delete all certificates in a station because user can send the GetInstalledCertificateIdsRequest to get only one or few GetCertificateIdUseEnumTypes, e.g., V2GRootCertificate. How about finding all certificate types inside the `certificateHashDataList` at first and only delete these types of certificates?

Reply:
for now adjusting module to not automatically trigger GetInstalledCertificateIds, and when GetInstalledCertificateIds payload comes in from the charger, deleting only installed certificate records with matching certificate types.